mono_repo/lib/src/commands/github/github_yaml.dart

```diff
@@ -366,39 +372,64 @@ Map<String, dynamic> _githubJobYaml(
             },
           ),
         packageFlavor.configurationMap(sdkVersion),
-        {
-          'id': 'checkout',
-          'uses': ActionInfo.checkout.usesValue,
-        },
-        for (var command in runCommands) command.runContent,
+        ..._beforeSteps(runCommands.whereType<_CommandEntry>()),
+        ActionInfo.checkout.usage(
+          name: 'Checkout repository',
+          id: 'checkout',
+        ),
+        for (var command in runCommands) ...command.runContent,
       ],
-    };
+    );
+
+Set<TaskType> _orderedTypes(Iterable<_CommandEntry> commands) =>
+    SplayTreeSet.of(commands.map((e) => e.type).whereType<TaskType>());
 
-class _CommandEntry {
+Iterable<Step> _beforeSteps(
+  Iterable<_CommandEntry> commands,
+) sync* {
+  for (var type in _orderedTypes(commands)) {
+    yield* type.beforeAllSteps;
+  }
+}
+
+class _CommandEntryBase {
   final String name;
   final String run;
+
+  _CommandEntryBase(this.name, this.run);
+
+  /// The entry in the GitHub Action stage representing this object.
+  ///
+  /// See https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idsteps
```

Comment:
This comment isn't really relevant now, it should move to where we are actually producing the yaml (or map?)